packages/ui-components/src/components/DateTimePicker/DateTimePicker.component.tsx

```diff
@@ -0,0 +1,651 @@
+  /** The name of the DateTimePicker input element */
+  name?: string
+  /** Set to `true` to not display a calendar at all. To create a time picker, set `enableTime` to true, too. */
+  noCalendar?: boolean
+  /** A handler to be executed when the DateTimePicker input element looses focus. */
+  // eslint-disable-next-line no-unused-vars
+  onBlur?: (date?: Date[], dateStr?: string) => void
+  /** A handler to be executed when the selected date(s), date range or time changes */
+  // eslint-disable-next-line no-unused-vars
+  onChange?: (date?: Date[], dateStr?: string) => void
+  /** A handler to be executed when the DateTimePicker value is reset by clicking the clear icon. The onChnage handler will be fired in this event too, onClear is more specific. */
+  // eslint-disable-next-line no-unused-vars
+  onClear?: (date?: Date[], dateStr?: string) => void
+  /** A handler to be executed when the DateTimePicker calendar closes */
+  // eslint-disable-next-line no-unused-vars
+  onClose?: (date?: Date[], dateStr?: string) => void
+  /** A handler to be executed when the DateTimePicker input element receives focus. */
+  // eslint-disable-next-line no-unused-vars
+  onFocus?: (date?: Date[], dateStr?: string) => void
+  /** A handler to be executed when the selected month changes */
+  // eslint-disable-next-line no-unused-vars
+  onMonthChange?: (date?: Date[], dateStr?: string) => void
+  /** A handler to be executed when the DateTimePicker calendar opens */
+  // eslint-disable-next-line no-unused-vars
+  onOpen?: (date?: Date[], dateStr?: string) => void
+  /** A handler to be executed when the DateTimePicker component is ready */
+  // eslint-disable-next-line no-unused-vars
+  onReady?: (date?: Date[], dateStr?: string) => void
+  /** A handler to be executed when the selected year changes */
+  // eslint-disable-next-line no-unused-vars
+  onYearChange?: (date?: Date[], dateStr?: string) => void
+  /** The placeholder of the DateTimePicker input element */
+  placeholder?: string
+  /** Whether the DateTimePicker should be marked as required. Requires a `Label` to be set. */
+  required?: boolean
+  /** Whether the current month in the date picker should be displayed as shorthand, e.g. "Jan" instead of "January" */
+  shorthandCurrentMonth?: boolean
+  /** The number of months to show in the date picker */
+  showMonths?: number
+  /** A text to render when the DateTimePicker was successfully validated */
+  successtext?: React.ReactNode
+  /** Displays time picker in 24 hour mode without AM/PM selection when enabled. Requires `enableTime` to be set, too. Default is `false`. */
+  time_24hr?: boolean
+  /** Whether the DateTimePicker has been successfully validated */
+  valid?: boolean
+  /** The value of the datepicker. Date Objects, timestamps, ISO date strings, chronological date strings `YYYY-MM-DD HH:MM` (must be compatible to current `dateFormat`), and the shortcut `today` are all accepted. */
+  value?: DateOption | DateOption[]
```

Comment:
**(Breaking Change)**

Please also have a look at this

Reply:
Updated type

Reply:
Here, value should accept the types string | number | object | array. This is necessary because the value is transformed into a string within the component:
```js
const stringifiedValue = JSON.stringify(value)
```